include/api_manager/request.h

```diff
@@ -44,6 +44,10 @@ class Request {
   // It should be "Unparsed" original URL path.
   virtual std::string GetUnparsedRequestPath() = 0;
 
+  // Update r_->uri, r_->request_line, r_->unparsed_uri based on
+  // unparsed_request_path
+  virtual void SetRequestPath(const std::string &unparsed_request_path) = 0;
```

Comment:
we may have to make api_base_path as global of whole proxy,  not per api, nor per service config.   Since users can only set it from start_esp argument,  it is easier to make to global.

If we go this way,  I prefer this approach:
api_manager interface,  add a use function call

string ReWriteURL(const string& url) = 0;

At the start of the each call, at our filter_handler,  Nginx call this function.  That is it.